### DISCUSSION TITLE
Parsing to bools and enums

### DISCUSSION THREAD

**Discussion by jgopel** [foonathan/lexy#125] [Q&A] [answered]
I'm trying to build a compile-time parser for [graphviz DOT](https://graphviz.org/doc/info/lang.html). I'm working on the intro stanza, which is `[strict] (graph|digraph)`. I would like to represent the graph type as an enum and strictness as a bool. I cannot find a way to get lexy to parse to those types or to handle the optional-ness of `strict`.

Sample code for what I'm trying to do with tests, etc: https://godbolt.org/z/frxjKMbda

**Comment by jgopel** [comment]
I suspect the answer to these questions lies somewhere in the docs or examples, but I haven't been able to craft the right search to find them - my apologies if I've overlooked something.

**Comment by foonathan** [comment, marked as answer]
For booleans, you can use [`dsl::flag`](https://lexy.foonathan.net/reference/dsl/flags/#flag), for enums, you can use [`dsl::symbol`](https://lexy.foonathan.net/reference/dsl/symbol/): https://godbolt.org/z/qrd3YK8cx
